### PR TITLE
fix(Core/Script): Thorim Hardmode & Sif Frostbolt Volley

### PR DESCRIPTION
**Pull request by disclosurez**
Sif's frostbolt volley on Thorim hardmode is currently broken and triggers too often, this will fix the issue with the boss.
This is an old issue that I paid for from an a dev external to AzerothCore

Fix Frostbolt Volley For Sif On Thorim Hardmode - Ulduar

## CHANGES PROPOSED:
-  Add proper stage to Sif on Thorim Hardmode In Ulduar

## ISSUES ADDRESSED:
- Closes 
Sif's frostbolt volley on Thorim hardmode is currently broken and triggers to often, this will fix the issue with the boss.

## TESTS PERFORMED:
- Confirmed working with multiple raids with a live production server of ~50 online players

## HOW TO TEST THE CHANGES:
--> Engage in Thorim Hardmode, and notice the difference with frostbolt volleys no longer casting too often.

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp [hunk 1]
```diff
@@ -905,9 +906,15 @@ class boss_thorim_sif : public CreatureScript
...
                 case EVENT_SIF_FROSTBOLT_VALLEY:
-                    me->CastSpell(me, SPELL_FROSTBOLT_VALLEY, false);
+					_allowCast = false;
+                    events.ScheduleEvent(EVENT_SIF_STAGE_TWO, 1500);
                     events.RepeatEvent(13000);
                     return;
+                case EVENT_SIF_STAGE_TWO:
+                    me->CastSpell(me, SPELL_FROSTBOLT_VALLEY, false);
+					_allowCast = true;
+					events.PopEvent();
```
Comment (Helias) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp:916]: fix indentations, convert tab in 4 spaces to be compliant

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp [hunk 2]
```diff
@@ -905,9 +906,15 @@ class boss_thorim_sif : public CreatureScript
                     events.ScheduleEvent(EVENT_SIF_BLIZZARD, 15000);
                     break;
                 case EVENT_SIF_FROSTBOLT_VALLEY:
-                    me->CastSpell(me, SPELL_FROSTBOLT_VALLEY, false);
+					_allowCast = false;
```
Comment (Helias) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp:909]: same here

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp [hunk 3]
```diff
@@ -234,6 +234,7 @@ enum ThorimEvents
 
     EVENT_DR_COMMONER_PM                    = 140,
     EVENT_DR_COMMONER_LB                    = 141,
+	EVENT_SIF_STAGE_TWO                    = 142,
```
Comment (Helias) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp:237]: same here

**Comment by BarbzYHOOL** [comment]
>  This is an old issue that I paid for from an a dev external to AzerothCore

thank you disclosurez!

I fixed the spaces for you because I'm already thankful for you pushing this fix!

**Comment by BarbzYHOOL** [comment]
I dunno for the code logic, don't even know the boss, but "SIF_EVENT_PHASE_TWO" doesn't sound on par with the other similar variables, but maybe semantically it's good as it effectively describes a phase 2, no idea, need someone who knows wotlk

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp [hunk 4]
```diff
@@ -905,9 +906,15 @@ class boss_thorim_sif : public CreatureScript
                     events.ScheduleEvent(EVENT_SIF_BLIZZARD, 15000);
                     break;
                 case EVENT_SIF_FROSTBOLT_VALLEY:
-                    me->CastSpell(me, SPELL_FROSTBOLT_VALLEY, false);
+                    _allowCast = false;
+                    events.ScheduleEvent(EVENT_SIF_STAGE_TWO, 1500);
                     events.RepeatEvent(13000);
                     return;
+                case EVENT_SIF_STAGE_TWO:
+                    me->CastSpell(me, SPELL_FROSTBOLT_VALLEY, false);
```
Comment (gengarshadowball) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp:914]: Can you change the variable name to volley instead? Just for consistency 😄

Comment (BarbzYHOOL) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp:914]: it's written valley lol, it has a typo by default, right?

Comment (gengarshadowball) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_thorim.cpp:914]: Yes, must have been someone coding till 5 am 😪

**Comment by Racknar0** [comment]
image

**Comment by Voxstrasza** [comment]
> 
> 
> I dunno for the code logic, don't even know the boss, but "SIF_EVENT_PHASE_TWO" doesn't sound on par with the other similar variables, but maybe semantically it's good as it effectively describes a phase 2, no idea, need someone who knows the boss

@BarbzYHOOL Sif is only active on Thorim if you complete the gauntlet within the time limit.  Phase 2 happens regardless (Thorim jumping down and attacking players), but Sif will only be casting frostbolt and frostbolt volley if you do the hardmode timer.

**Review by Si1ker — COMMENTED** [review]
switch (events.GetEvent())  => switch (events.ExecuteEvent())

All 'events.PopEvent();'  have been deleted.

**Comment by pklloveyou** [comment]
Tested

**Comment by FrancescoBorzi** [comment]
@disclosurez can you update this with latest AC master ?

**Comment by Si1ker** [comment]
Useless change.
If the author wants to re-open, it should include proof/evidence for this change.
Closing PR.

**Pull request closed without merging**